packages/test-utils/src/utils.ts

```diff
@@ -101,6 +147,59 @@ export const waitForCondition = async (
     })
 }
 
+/**
+ * Run functions and wait conditions to become true by re-evaluating every `retryInterval` milliseconds. Returns a promise created with Promise.all() 
+ * and waitForCondition() calls. Calls the functions after creating the promise.
+ * 
+ * @param operations function(s) to call
+ * @param conditions condition(s) to be evaluated; condition functions should return boolean or Promise<boolean> and have
+ * no side-effects.
+ * @param timeout amount of time in milliseconds to wait for
+ * @param retryInterval how often, in milliseconds, to re-evaluate condition
+ * @param onTimeoutContext evaluated only on timeout. Used to associate human-friendly textual context to error.
+ * @returns {Promise<unknown[]>} resolves immediately if
+ * conditions evaluate to true on a retry attempt within timeout. If timeout
+ * is reached with conditionFn never evaluating to true, rejects.
+ */
+export const runAndWaitForConditions = (
```

Comment:
Should be `async` for same reasons above.
```suggestion
export const runAndWaitForConditions = async (
```